### ISSUE TITLE
Hazelcast New Client doesn't support RingBuffer

### ISSUE THREAD

**Issue by gAmUssA** [hazelcast/hazelcast#6358]
Hi team,
@pveentjer @sancar @asimarslan 

Hazelcast New Client doesn't support RingBuffer.

```
Exception in thread "main" java.lang.UnsupportedOperationException
	at com.hazelcast.client.impl.HazelcastClientProxy.getRingbuffer(HazelcastClientProxy.java:83)
	at demo.Producer.main(Producer.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
```

Hazelcast version 3.5.2
Both OSS and Enterprise versions are affected.

Cheers,
Vik

**Comment by gAmUssA** [comment]
Just checked with the latest 3.6-SNAPSHOT from https://oss.sonatype.org/content/repositories/snapshots. Looks like RingBuffer works for new client as well.

Team, please, confirm and close the issue.

**Comment by sancar** [comment]
It is available in 3.6-SNAPSHOT  but not backported to maintenance branch, although it seems it is backported to maintenance branch for old client. 
@pveentjer was that intentional ?

**Comment by enesakar** [comment]
we will support it starting from 3.6
